fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/type/NumberParmBehaviorUtil.java

```diff
@@ -0,0 +1,289 @@
+            int loc = t.indexOf('E');
+            if (loc > -1) {
+                factor = -1 * (loc - 2);
+            }
+        }
+
+        BigDecimal scaleFactor = new BigDecimal("5e" + -1 * factor);
+        return original.subtract(scaleFactor);
+    }
+
+    public static BigDecimal generateUpperBound(BigDecimal original) {
+        int factor = calculateSignificantFigures(original);
+        if (factor < 0) {
+            factor = Math.abs(factor) - 1;
+        } else {
+            factor = original.scale() + 1;
+
+            // Mutate it to the ORIGINAL string
+            String t = "" + original;
+            int loc = t.indexOf('E');
+            if (loc > -1) {
+                factor = -1 * (loc - 2);
+            }
+        }
+        BigDecimal scaleFactor = new BigDecimal("5e" + -1 * factor);
+        return original.add(scaleFactor);
+    }
+
+    /**
+     * calculates the significant figures
+     * based on
+     * <a href="https://en.wikipedia.org/wiki/Significant_figures">Significant
+     * Figures</a>
+     * 
+     * @param original
+     * @return
+     */
+    public static int calculateSignificantFigures(BigDecimal original) {
+        int count = original.precision();
+        if (original.scale() <= 0) {
+            // Common pattern is to strip zeros... <code>.stripTrailingZeros()</code> 
+            // We don't per the pattern in FHIR. 
+            count += original.scale();
+        }
+        return count;
+    }
+
+    /**
+     * per the specification we DO NOT process EB/SA integers.
+     * <a href="https://hl7.org/fhir/search.html#prefix">FHIR Specification: Search
+     * - Prefixes</a>
+     * <code> 
+     * sa (starts-after) and eb (ends-before) are not used with integer values but are used for decimals. 
+     * </code>
+     * 
+     * @param prefix
+     * @param resourceType
+     * @param queryParm
+     * @param originalNumber
+     * @return boolean indicating that an integer search is being run.
+     * @throws FHIRPersistenceException
+     */
+    public static boolean checkIntegerSearchWithSaEb(Prefix prefix, Class<?> resourceType, Parameter queryParm,
+            BigDecimal originalNumber)
+            throws FHIRPersistenceException {
+        boolean isIntegerSearch = false;
+        try {
+            isIntegerSearch = ValueTypesFactory.getValueTypesProcessor().isIntegerSearch(resourceType, queryParm);
+        } catch (FHIRSearchException e) {
+            log.log(Level.INFO, "Caught exception while checking the value types for parameter '"
+                    + queryParm.getCode() + "'; continuing...", e);
+            // do nothing
+        }
+
+        if (isIntegerSearch) {
+            if (prefix == Prefix.EB || prefix == Prefix.SA) {
+                throw new FHIRPersistenceException(
+                        "Search prefixes '" + Prefix.EB.value() + "' and '" + Prefix.SA.value()
+                                + "' are not supported for integer searches.");
```

Comment:
this is consistent with the previous impl, although I don't really think it matters.  i guess its fine to keep

Reply:
Yeah - I kept it for that reason.